src/pyqasm/algorithms/solovay_kitaev/generator.py

```diff
@@ -0,0 +1,102 @@
+import pickle
+import sys
+from collections import deque
+
+import numpy as np
+
+gate_sets = {
```

Comment:
Can we instead expand the `IDENTITY_WEIGHT_GROUP` to also contain the matrix and use that here as I see duplication between the two data structures